src/GraphQL/Types/Scalars/Enumeration/CaseInsensitiveEnumValues.cs

```diff
@@ -0,0 +1,44 @@
+using GraphQLParser;
+
+namespace GraphQL.Types;
+
+/// <summary>
+/// A class that represents a case insensitive set of enumeration definitions.
+/// </summary>
+public class CaseInsensitiveEnumValues : EnumValues
+{
+    private readonly IDictionary<string, EnumValueDefinition> _upperValues = new Dictionary<string, EnumValueDefinition>();
+    private readonly IDictionary<string, EnumValueDefinition> _constantCaseValues = new Dictionary<string, EnumValueDefinition>();
+
+    /// <inheritdoc />
+    public override void Add(EnumValueDefinition value)
+    {
+        base.Add(value);
+        _upperValues[value.Name.ToUpperInvariant()] = value;
+        _constantCaseValues[value.Name.ToConstantCase()] = value;
```

Comment:
I think you can use a single dictionary for both keys. Moreover, for a single-word enum, both values will produce the same key.

Reply:
> I think you can use a single dictionary for both keys. Moreover, for a single-word enum, both values will produce the same key.

Good point